### PR TITLE
Fix AWS_REGION and add AWS_CA_BUNDLE to s3_secrets.go

### PR DESCRIPTION
**Pull request by elukey**
The boto3 package seems to look for AWS_DEFAULT_REGION, meanwhile
kfserving uses AWS_REGION, that doesn't seem to be working
(tested following
https://github.com/kubeflow/kfserving/blob/master/docs/samples/storage/s3/README.md).

This change doesn't replace all occurrences of AWS_REGION, but only
the ones that set the environment variables.

GH-1765

Add also the possibility to force boto to use a custom CA Bundle path when validating the TLS certificate of the S3 endpoint. Our use case is related to using an internal S3 endpoint (based on Openstack Swift) offering a TLS certificate signed by our internal CA.

GH-1766

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by elukey** [comment]
Hi everybody, any feedback?

File: python/kfserving/kfserving/api/creds_utils.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def set_s3_credentials(namespace, credentials_file, service_account,
                               is specified, will attach created secret with the service account,
                               otherwise will create new one and attach with created secret.
         s3_endpoint(str): S3 settings variable S3_ENDPOINT.
-        s3_region(str): S3 settings variable AWS_REGION.
+        s3_region(str): S3 settings variable AWS_DEFAULT_REGION.
```
Comment (yuzisun) [inline python/kfserving/kfserving/api/creds_utils.py:70]: Also need to add the s3_cabundle argument here?

File: pkg/credentials/s3/s3_secret.go [hunk 2]
```diff
@@ -49,6 +50,7 @@ var (
 	InferenceServiceS3SecretHttpsAnnotation      = constants.KFServingAPIGroupName + "/" + "s3-usehttps"
 	InferenceServiceS3UseVirtualBucketAnnotation = constants.KFServingAPIGroupName + "/" + "s3-usevirtualbucket"
 	InferenceServiceS3UseAnonymousCredential     = constants.KFServingAPIGroupName + "/" + "s3-useanoncredential"
+	InferenceServiceS3CABundlePathAnnotation     = constants.KFServingAPIGroupName + "/" + "s3-cabundlepath"
```
Comment (yuzisun) [inline pkg/credentials/s3/s3_secret.go:53]: `s3-cabundlepath` -> `s3-cabundle` just to be consistent with the env variable name?

**Comment by yuzisun** [comment]
> Hi everybody, any feedback?

Thanks @elukey ! The change mostly lgtm, Can you help link the AWS S3 doc regarding the region variable so we can have a reference?

File: python/kfserving/kfserving/api/creds_utils.py (hunk 1 above)
Comment (elukey) [reply python/kfserving/kfserving/api/creds_utils.py:70]: Totally missed this, going to follow up!

File: pkg/credentials/s3/s3_secret.go (hunk 2 above)
Comment (elukey) [reply pkg/credentials/s3/s3_secret.go:53]: I was unsure about naming, will change it :)

**Comment by elukey** [comment]
> > Hi everybody, any feedback?
> 
> Thanks @elukey ! The change mostly lgtm, Can you help link the AWS S3 doc regarding the region variable so we can have a reference?

Sure! Do you want only a reference in here or do you prefer also a reference in a code's comment? 

AWS Cli: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

**Comment by yuzisun** [comment]
> > > Hi everybody, any feedback?
> > 
> > 
> > Thanks @elukey ! The change mostly lgtm, Can you help link the AWS S3 doc regarding the region variable so we can have a reference?
> 
> Sure! Do you want only a reference in here or do you prefer also a reference in a code's comment?
> 
> AWS Cli: https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html
> Boto: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/configuration.html#using-environment-variables

code comments will be helpful, thanks!

**Comment by elukey** [comment]
The test to upload something to AWS s3 fails for some reason, will dig a bit more into it. Tried to modify the tests but didn't succeed. Any suggestion is welcome :)

**Comment by elukey** [comment]
I see that in https://github.com/kubeflow/testing/search?q=AWS_REGION the variable is mentioned a lot, so this may explain why I am struggling with tests. But boto doesn't support it, more info: https://github.com/boto/boto3/issues/2574

**Comment by elukey** [comment]
The error that I see is:

```
 INFO|2021-08-31T10:01:27|/src/kubeflow/testing/py/kubeflow/testing/argo_client.py|48| Parsing ApiException body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Unauthorized","reason":"Unauthorized","code":401}
INFO|2021-08-31T10:01:27|/src/kubeflow/testing/py/kubeflow/testing/argo_client.py|60| ApiException code=401 
```

Initially I thought that it was related to the s3 api, but maybe it is something (un)related to the k8s api?

**Comment by elukey** [comment]
@yuzisun I found the same error mentioned above in other pull requests, is it a known issue?

**Comment by elukey** [comment]
> > @yuzisun I found the same error mentioned above in other pull requests, is it a known issue?
> 
> ye might be due to the auth token expiration which throws the authorization error, rerunning the test.

Still seeing the issue in the last presubmit job :(

**Comment by elukey** [comment]
Created https://github.com/kubeflow/kfserving/issues/1788

**Comment by yuzisun** [comment]
> Created #1788

@elukey we found the issue, the image built from CI is conflicting with the PR from kserve repo.. see https://github.com/kserve/kserve/pull/44

**Comment by elukey** [comment]
> > Created #1788
> 
> @elukey we found the issue, the image built from CI is conflicting with the PR from kserve repo.. see [kserve/kserve#44](https://github.com/kserve/kserve/pull/44)

Thanks a lot! I've re-run the test but it still fails :(

**Comment by elukey** [comment]
@yuzisun this time it worked!

**Comment by yuzisun** [comment]
Thanks @elukey ! 
/lgtm
/approve

**Comment by umka1332** [comment]
Hi @elukey,

How do you put the CA bundle file(s) inside storage-initializer container? Is it built into image or were you able to mount it somehow into the container? If the second, could you please share steps to do this? I've tried a lot of things but non of them works with default image.

**Comment by elukey** [comment]
@umka1332 Hi! Correct yes, we build our own versions of the Kserve docker images, including the bundle. Then we simply add the environment variables and that's it :)

**Pull request merged**